test/awss3.jl

```diff
@@ -0,0 +1,47 @@
+function minio_server(body, dirs=[mktempdir()]; address="localhost:9005")
+    server = Minio.Server(dirs; address)
+
+    try
+        run(server; wait=false)
+        sleep(0.5)  # give the server just a bit of time, though it is amazingly fast to start
+
+        config = MinioConfig(
+            "http://$address"; username="minioadmin", password="minioadmin"
+        )
+        body(config)
+    finally
+        # Make sure we kill the server even if a test failed.
+        kill(server)
+    end
+end
+
+@testset "AWSS3 usage" begin
+    minio_server() do config
+        s3_create_bucket(config, "test-bucket")
+
+        file_format = "lpcm.zst"
+        file_path = S3Path("s3://test-bucket/prefix/samples.$(file_format)"; config)
+        recording_uuid = uuid4()
+        start = Second(0)
+
+        info = SamplesInfoV2(sensor_type="eeg",
+            channels=["a", "b"],
+            sample_unit="unit",
+            sample_resolution_in_unit=1.0,
+            sample_offset_in_unit=0.0,
+            sample_type=Int16,
+            sample_rate=100.0)
+        samples = Samples(zeros(sample_type(info), 2, 300), info, true)
+
+        signal = Onda.store(file_path, file_format, samples, recording_uuid, start)
+        @test signal.file_path isa S3Path
+
+        loaded_samples = Onda.load(signal; encoded=true)
+        @test samples == loaded_samples
+
+        # Load subspan to exercise method
+        span = TimeSpan(0, Second(1))
+        loaded_span = Onda.load(signal, span; encoded=true)
+        @test loaded_samples[:, span] == loaded_span
```

Comment:
for some reason it doesn't seem like this has hit that added method, at least according to the "missed coverage" annotations above.  I wonder if somehow the package extension isn't getting loaded properly in tests?

we also have some tests in OndaBatches.jl which try to really make sure we're actually passing the byte range to AWSS3: https://github.com/beacon-biosignals/OndaBatches.jl/blob/a43b6f0a6837c2a478a1845c69728c1851e234c0/test/utils.jl#L22-L27

Reply:
locally I confirmed it was loading, but I agree something seems off here. Weirdly codecov says the other method WAS hit in the extension code...